test/NinjaTools.FluentMockServer.API.Tests/Proxy/Matchers/EvaluatorTests.cs

```diff
@@ -0,0 +1,261 @@
+                    Cookies = new Dictionary<string, string>
+                    {
+                        {"abcd", "application/json"}
+                    },
+                    Query =  "?id=0",
+                    Method = "GET",
+                    Headers = new Dictionary<string, string[]>
+                    {
+                        { "a", new [] {""}}
+                    },
+                    BodyMatcher = new RequestBodyMatcher
+                    {
+                        Content = "?"
+                    }
+                };
+                var evalA = Eval(0, false);
+                var ctxA = GetContext(r =>
+                {
+                    r.Method = HttpMethods.Delete;
+
+                });
+                Add(reqA, ctxA, evalA);
+            }
+
+            private void Data2()
+            {
+                var reqA = new RequestMatcher
+                {
+                    Path = "/x&/y(",
+                    Query = "?id=123"
+                };
+                var evalA = Eval(6, false);
+                var ctxA = GetContext(r => { });
+
+                Add(reqA, ctxA, evalA);
+            }
+        }
+    }
+
+
+
+    public class EvaluatorTests: XUnitTestBase<EvaluatorTests>
+    {
+        /// <inheritdoc />
+        public EvaluatorTests(ITestOutputHelper output) : base(output)
+        { }
+
+        [Theory]
+        [ClassData(typeof(EvaluatorTestData.InvalidRequestMatcherTestData))]
+        public void Should_Return_EvaluationUnsuccessfulResult_When_Not_Matching(RequestMatcher matcher, HttpContext context, EvaluationResultBase expectedResult)
+        {
+            // Act && Assert
+            var visitor = new RequestEvaluatorVistor(new EvaluationContext(context));
+            var result = visitor.Evaluate();
+            Dump(matcher, "matcher");
+            result.Should().As<EvaluationResultBase>().Should().Be(false);
+            result.IsMatch.Should().BeFalse();
+            result.Score.Should().Be(expectedResult.Score);
+        }
+
+        [Theory]
+        [ClassData(typeof(EvaluatorTestData.ValidRequestMatcherTestData))]
+        public void Should_Return_EvaluationSuccessfulResult_When_Matching(RequestMatcher matcher, HttpContext context, EvaluationResultBase expectedResult)
+        {
+            // Act && Assert
+            var visitor = new RequestEvaluatorVistor(new EvaluationContext(context));
+            var result = matcher.Accept(() => visitor);
+            Dump(matcher, "matcher");
+            result.Should().As<EvaluationResultBase>().Should().Be(false);
+            result.IsMatch.Should().Be(true);
+            result.Score.Should().Be(expectedResult.Score);
+        }
+
+        //
+        // [Theory]
+        // [MemberData(nameof(GetScoreTestData))]
+        // public void Score_Should_Be_Higher_For_The_Matcher_Who_Is_Closer_To_The_HttpRequest(RequestMatcher more, HttpContext context, RequestMatcher less)
+        // {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)